aha-table.html

```diff
@@ -635,6 +635,13 @@
       // with filtering/sorting, it's best just to go back to the first page
       this.$.pagination.goToPageNumber(1);
       this._updateDisplayedRows();
+
+      //update selected rows when changing the filter if the select all is enabled
+      selectAllCheckbox = Polymer.dom(this.root).querySelector("#selectAllCheckbox");
+      if(selectAllCheckbox !== null && selectAllCheckbox !== undefined){
+        if(selectAllCheckbox.checked)
+          this._setAllRows(true);
```

Comment:
I don't think you actually want to set them all true as they change/remove their filtered value, you just want to uncheck the selectAll checkbox:

```
if(selectAllCheckbox !== null && selectAllCheckbox !== undefined && selectAllCheckbox.checked) {
          selectAllCheckbox.checked = false;
      }
```